pmd-test/src/main/java/net/sourceforge/pmd/cli/BaseCLITest.java

```diff
@@ -31,7 +31,9 @@
 /**
  * @author Romain Pelisse &lt;belaran@gmail.com&gt;
  *
+ * @deprecated Only pmd-cli module should use / test the CLI.
```

Comment:
Maybe we need a pmd-cli-integration-tests module to test with actual language modules? I would prefer pmd-cli not to depend on language modules, even if it's only a test-scoped dependency. Otherwise pmd-dist could contain those tests I guess

Reply:
If we were to do this, I'd do it in pmd-dist (as the same would apply to Ant).

However, I don't think this is particularly pressing… the way both CLI and Ant are being setup with separate modules, they are simply interfaces into `PMDConfiguration` / `PmdAnalysis`. These modules can be sufficiently tested on their own with the Dummy Languages. They could even be put into independent repositories without issue (like we do for the Eclipse Plugin as another "interface into `PMDConfiguration` / `PmdAnalysis`").

The only real integration point for language modules is ensuring the `LanguageRegistry` is properly being able to detect it (that is, that the service definitions are properly setup), and that can be tested from the language modules themselves, as it only requires accessing pmd-core (and this is something we should start doing…).

The current usages for this `BaseCLITest` as it stands are:
- issues with pmd-core itself, irrespective of CLI / Ant, or even the specific language (ie: [these tests](https://github.com/pmd/pmd/blob/master/pmd-java/src/test/java/net/sourceforge/pmd/cli/CLITest.java#L76-L107) from pmd-java are actually testing [an issue with pmd-core's `RuleSetFactory` not emitting errors](https://sourceforge.net/p/pmd/bugs/1231/).
- testing behavior of the `LanguageVersionDiscoverer` irrespective of the language in use (ie: [these tests](https://github.com/pmd/pmd/blob/master/pmd-xml/src/test/java/net/sourceforge/pmd/lang/xml/XmlCliTest.java#L44-L54) in pmd-xml).
- integration tests with very little value add that limit themselves to checking the run didn't blow up.

Reply:
We actually have some basic tests for all languages in pmd-dist already, even for Ant (see `AntIT` and `AllRulesIT`).

For the JUnit5 migration (#3797), I skipped for now this BaseCLITest class. But maybe, we can change/move the tests, that we don't actually need separate CLI tests in each module, like @jsotuyod proposes. And we can delete this base test class... I'll keep that in mind, when I continue on JUnit5 migration.